lupa/_lupa.pyx

```diff
@@ -1399,12 +1402,13 @@ cdef bint call_python(LuaRuntime runtime, lua_State *L, py_object* py_obj) excep
 
 cdef int py_call_with_gil(lua_State* L, py_object *py_obj) with gil:
     cdef LuaRuntime runtime = None
+    runtime = <LuaRuntime?>py_obj.runtime
```

Comment:
The basic thing to remember is that raising an exception can actually fail and raise an exception, so if there is nothing around that can deal with that (e.g. because we are being called by Lua), then you'll have to be paranoid and wrap everything in bare try-excepts and try-finallys. Luckily, Cython can issue a warning if it finds a spot where an exception can be raised and it doesn't know how to propagate it.